modules/engine/src/animation/timeline.d.ts

```diff
@@ -0,0 +1,39 @@
+interface ChannelProps {
+  delay?: number
+  duration?: number
+  rate?: number
+  repeat?: number
+}
+
+interface Channel {
+  time: number
+  delay: number
+  duration: number
+  rate: number
+  repeat: number
+}
+
+interface Animation {
+  setTime: (time: number) => void
+}
+
+export class Timeline {
+  time: number
+  channels: Map<number, Channel>
+  animations: Map<number, Animation>
+  playing: boolean
+  lastEngineTime: number
+  constructor();
+  addChannel(props: ChannelProps): number;
```

Comment:
By using `ChannelProps` instead of actually typing out the props here, the user can't see in vscode hover info which props are actually available. So far I have always chosen to type out the values here. 

I realize sometime people want to reuse a type describing the arguments. However, all those small helper types then become part of the public interface and need to be documented and stable.

Reply:
I see the value of typing them out here, so yeah I'll give that a go. I do think there's value in reusing types where the implementation is strictly tying types together (e.g. any function that passes a param through to another function where is a type changes in one place it would need to change in the other to stay valid).

The catch 22 is I don't always know what is reused until I'm fart along with the typings, so how about I do a final pass to reduce these small single use interfaces? I agree that reducing nominal surface is good, since in reality everything is structurally typed with JS.